src/util/compiler_defs.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+/// Header to define some properties that are OS or Compiler specific
+
+#ifndef GNUC_PREREQ
+#if defined(__GNUC__) && defined(__GNUC_MINOR__) && defined(__GNUC_PATCHLEVEL__)
+#define GNUC_PREREQ(maj, min, patch)                                           \
+  ((__GNUC__ << 20) + (__GNUC_MINOR__ << 10) + __GNUC_PATCHLEVEL__ >=          \
+   ((maj) << 20) + ((min) << 10) + (patch))
+#elif defined(__GNUC__) && defined(__GNUC_MINOR__)
+#define GNUC_PREREQ(maj, min, patch)                                           \
+  ((__GNUC__ << 20) + (__GNUC_MINOR__ << 10) >= ((maj) << 20) + ((min) << 10))
+#else
+#define GNUC_PREREQ(maj, min, patch) 0
+#endif
+#endif
+
+#if __has_attribute(noinline) || GNUC_PREREQ(3, 4, 0)
+#define ATTRIBUTE_NOINLINE __attribute__((noinline))
+#elif defined(_MSC_VER)
+#define ATTRIBUTE_NOINLINE __declspec(noinline)
+#else
+#define ATTRIBUTE_NOINLINE
+#endif
+
+#if __has_attribute(used) || GNUC_PREREQ(3, 1, 0)
+#define ATTRIBUTE_USED __attribute__((__used__))
+#else
+#define ATTRIBUTE_USED
+#endif
+
+#if !defined(NDEBUG)
+#define DUMP_METHOD ATTRIBUTE_NOINLINE ATTRIBUTE_USED
+#else
+#define DUMP_METHOD ATTRIBUTE_NOINLINE
+#endif
+
+// Add your losing compiler to this list.
+#if !defined bool &&                                                           \
+  (defined __SUNPRO_CC && (__SUNPRO_CC < 0x500 || __SUNPRO_CC_COMPAT < 5) ||   \
+   defined __xlC__ || defined __DECCXX && __DECCXX_VER < 60000000 ||           \
+   defined _MSC_VER && _MSC_VER < 1100)
```

Comment:
I doubt that any compiler which doesn't support `bool` can successfully compile ESBMC. Can we remove this?

Reply:
Removed it